tck/api/src/main/java/org/eclipse/microprofile/metrics/tck/tags/TagsTest.java

```diff
@@ -157,4 +163,69 @@ public void histogramTagsTest() {
         assertThat("Histogram is not registered correctly", registry.getHistogram(histogramRedMID), notNullValue());
         assertThat("Histogram is not registered correctly", registry.getHistogram(histogramBlueMID), notNullValue());
     }
+
+    @Test
+    @InSequence(6)
+    public void nonMatchingTagTest() {
+        Tag tagForNames = new Tag("name", "Bill");
+
+        String counterName = "org.eclipse.microprofile.metrics.tck.TagTest.counter.mismatch.tags";
+
+        Counter counter = registry.counter(counterName);
+
+        try {
+            registry.counter(counterName, tagForNames);
+
+        } catch (Exception cause) {
+            assertThat(cause, is(Matchers.<Exception>instanceOf(IllegalArgumentException.class)));
+            return;
+        }
```

Comment:
Purely stylistic suggestion: Replace the `try-catch` with Junit 5's `assertThrows`:

```java
assertThrows(IllegalArgumentException.class, () -> registry.counter(counterName, tagForNames));
```

If you like this, same change for the two blocks below also.

Reply:
@tjquinno  Ah yes! I was intending to use that earlier. But the current juit is 4.12 I believe. I was intending to update the junit separately at another time and make updates then as a separate stream of issues. Tthere's other instances in the TCK where we try to check for an exception.

Reply:
Ah ha. I did not bother to check what the exact JUnit dependency was. Sorry about the noise.